include/openenclave/bits/sgx/sgxproperties.h

```diff
@@ -223,6 +231,54 @@ typedef struct _oe_sgx_enclave_properties
     FAMILY_ID,                                                            \
     ALLOW_DEBUG,                                                          \
     true,                                                                 \
+    0,                                                                    \
+    HEAP_PAGE_COUNT,                                                      \
+    STACK_PAGE_COUNT,                                                     \
+    TCS_COUNT)
+
+/**
+ * Defines the SGX2 properties for an enclave
+ *
+ * The enclave properties should only be defined once for all code compiled into
+ * an enclave binary. These properties can be overwritten at sign time by
+ * the oesign tool.
+ *
+ * @param[in] PRODUCT_ID ISV assigned Product ID (ISVPRODID) to use in the
+ * enclave signature
+ * @param[in] SECURITY_VERSION ISV assigned Security Version number (ISVSVN)
+ * to use in the enclave signature
+ * @param[in] EXTENDED_PRODUCT_ID ISV assigned Extended Product ID (ISVEXTPRODID)
+ *  to use in the enclave signature (SGX2 feature)
+ * @param[in] FAMILY_ID ISV assigned Product Family ID (ISVFAMILYID)
+ * to use in the enclave signature (SGX2 feature)
+ * @param[in] ALLOW_DEBUG If true, allows the enclave to be created with
+ * OE_ENCLAVE_FLAG_DEBUG and debugged at runtime
+ * @param[in] PF_GP_EXCEPTIONS If true, allows the enclave to capture #PF/#GP
+ * exceptions if CPU supports the feature. The setting is ignored otherwise
+ * (SGX2 feature)
+ * @param[in] HEAP_PAGE_COUNT Number of heap pages to allocate in the enclave
+ * @param[in] STACK_PAGE_COUNT Number of stack pages per thread to reserve in
+ * the enclave
+ * @param[in] TCS_COUNT Number of concurrent threads in an enclave to support
+ */
+ #define OE_SET_ENCLAVE_SGX2(                                             \
+    PRODUCT_ID,                                                           \
```

Comment:
follow OE style guidance for new macros.  use lower_snake_case for args

Reply:
Filed as a separated issue: https://github.com/openenclave/openenclave/issues/3998